plugins/modules/monitoring/pagerduty_user.py

```diff
@@ -0,0 +1,266 @@
+            if e.response.status_code == 429:
+                self._module.fail_json(
+                    msg="Failed to add %s due to reaching the limit of making requests %s" % (pd_name, e))
+
+    # delete a user account from PD
+    def delete_user(self, pd_user_id, pd_name):
+        try:
+            user_path = path.join('/users/', pd_user_id)
+            self._apisession.rdelete(user_path)
+
+        except PDClientError as e:
+            if e.response.status_code == 404:
+                self._module.fail_json(
+                    msg="Failed to remove %s as user was not found %s" % (pd_name, e))
+            if e.response.status_code == 403:
+                self._module.fail_json(
+                    msg="Failed to remove %s due to inability to review the requested resource within the API token %s" % (pd_name, e))
+            if e.response.status_code == 401:
+                # print out the list of incidents
+                pd_incidents = self.get_incidents_assigned_to_user(pd_user_id)
+                self._module.fail_json(msg="Failed to remove %s as user has assigned incidents %s : %s" % (pd_name, pd_incidents, e))
+            if e.response.status_code == 429:
+                self._module.fail_json(
+                    msg="Failed to remove %s due to reaching the limit of making requests %s" % (pd_name, e))
+
+    # get incidents assigned to a user
+    def get_incidents_assigned_to_user(self, pd_user_id):
+        incident_info = {}
+        incidents = self._apisession.list_all('incidents',
+            params={'user_ids[]':[pd_user_id]})
+
+        for incident in incidents:
+            incident_info = {
+                'title': incident['title'],
+                'key': incident['incident_key'],
+                'status': incident['status']
+            }
+        return incident_info
+
+    # add a user to a team/teams
+    def add_user_to_teams(self, pd_user_id, pd_team, pd_role):
+        updated_team = None
+        for team in pd_team:
+            team_info = self._apisession.find('teams', team, attribute='name')
+            if team_info is not None:
+                try:
+                    updated_team = self._apisession.rput('/teams/'+team_info['id']+'/users/'+pd_user_id, json={
+                        'role':pd_role
+                    })
+                except PDClientError:
+                    updated_team = None
+        return updated_team
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            access_token=dict(type='str', required=True, no_log=True),
+            pd_user=dict(type='str', required=True),
+            pd_email=dict(type='str', required=True),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            pd_role=dict(type='str', default='responder', required=False, choices=['global admin', 'manager', 'responder', 'observer', 'stakeholder', 'limited skateholder', 'restricted access']),
```

Comment:
```suggestion
            pd_role=dict(type='str', default='responder', choices=['global admin', 'manager', 'responder', 'observer', 'stakeholder', 'limited skateholder', 'restricted access']),
```
if it's not given, `required=False` is a default. So, please remove it from the other places as well